### PR TITLE
Add functionality for import of existing repos.

### PR DESCRIPTION
**Pull request by goudine**
This commit contains the necessary changes to the GeoGig project in
order for the plugin (PR submitted to GeoServer project) to succesfully
import existing PostGres and directory backed repos via the web API.

Signed-off-by: goudine <agoudine@boundlessgeo.com>

### CODE REVIEW

File: src/web/api/src/main/java/org/locationtech/geogig/web/api/CommandBuilder.java [hunk 1]
```diff
@@ -95,6 +95,7 @@
         MAPPINGS.put("export", Export::new);
         MAPPINGS.put("export-diff", ExportDiff::new);
         MAPPINGS.put("import", Import::new);
+        MAPPINGS.put("importExistingRepo", Import::new);
```
Comment (jdgarrett) [inline src/web/api/src/main/java/org/locationtech/geogig/web/api/CommandBuilder.java]: This shouldn't be here, the command is handled in the geoserver plugin.

File: src/web/functional/src/test/java/org/geogig/web/functional/FunctionalTestContext.java [hunk 2]
```diff
@@ -57,7 +57,7 @@
     /**
      * A temporary folder to use for tests
      */
-    protected TemporaryFolder tempFolder = null;
+    public TemporaryFolder tempFolder = null;
```
Comment (jdgarrett) [inline src/web/functional/src/test/java/org/geogig/web/functional/FunctionalTestContext.java]: There is a function in this class called getTempFolder() that the code in the plugin can use.

File: src/web/api/src/main/java/org/locationtech/geogig/web/api/CommandBuilder.java (hunk 1 above)
Comment (jdgarrett) [inline src/web/api/src/main/java/org/locationtech/geogig/web/api/CommandBuilder.java]: This shouldn't be here, the command is handled on the geoserver plugin side.

File: src/web/api/src/main/java/org/locationtech/geogig/rest/repository/CommandResource.java [hunk 3]
```diff
@@ -277,7 +263,7 @@ protected MediaType resolveFormat(Form options, Variant variant) {
 
         private Function<MediaType, Representation> representation;
 
-        RestletContext(Repository geogig, Request request) {
+        public RestletContext(Repository geogig, Request request) {
```
Comment (jdgarrett) [inline src/web/api/src/main/java/org/locationtech/geogig/rest/repository/CommandResource.java]: The changes in this class are not utilized by the PR on the geoserver plugin.

**Comment by goudine** [comment]
Issued new PR to account for the code which was refactored

**Pull request closed without merging**